Backend/Interview.Domain/Rooms/Service/RoomService.cs

```diff
@@ -164,6 +165,59 @@ public async Task<IPagedList<RoomPageDetail>> FindPageAsync(
         });
     }
 
+    public async Task<List<RoomCalendarItem>> GetCalendarAsync(RoomCalendarRequest filter, CancellationToken cancellationToken = default)
+    {
+        var currentUserId = _currentUserAccessor.UserId;
+
+        var queryable = _db.Rooms
```

Comment:
It might be easier to go from the RoomParticipant table, after which take just the rooms. Probably less records would have to be scanned.

```c#
_db.RoomParticipant.AsNoTracking().Include(e => e.Room).Where(e => e.UserId == currentUserId).Select(e => e.Room)
```

Also be sure to add “AsNoTracking” when reading data from context when it will not change (then change tracking is turned off and such a query works faster)